hybris/egl/egl.c

```diff
@@ -488,8 +488,9 @@ static EGLImageKHR _my_eglCreateImageKHR(EGLDisplay dpy, EGLContext ctx, EGLenum
 
 	struct egl_image *image;
 	image = malloc(sizeof *image);
+	image->ws_dpy = hybris_egl_display_get_mapping(dpy);
+	image->ws_buffer = newbuffer;
 	image->egl_image = eik;
-	image->egl_buffer = buffer;
 	image->target = target;
 
 	return (EGLImageKHR)image;
```

Comment:
This cast is not valid because of the changes you made to egl_image struct, the EGLImageKHR entry needs to be first for this to be valid.